includes/components/class-attribute.php

```diff
@@ -1721,6 +1725,68 @@ public function add_meta_boxes( $meta_boxes ) {
 				// Add meta box.
 				$meta_boxes[ $model . '_' . $meta_box_name ] = $meta_box;
 			}
+
+			// Set model category settings key.
+			$category_key = $model . '_category_settings';
+
+			if ( ! $this->models[ $model ]['searchable'] || ! in_array( $category_key, array_keys( $meta_boxes ), true ) ) {
+				continue;
+			}
+
+			// Create sort form.
+			$sort_form = hp\create_class_instance( '\HivePress\Forms\\' . $model . '_sort' );
+
+			// Check sort form.
+			if ( ! $sort_form ) {
+				continue;
+			}
+
+			// Get sort options.
+			$model_options = $sort_form->get_fields()['_sort']->get_arg( 'options' );
+
+			// Add category options.
+			$category_options = [];
+
+			if ( isset( $_REQUEST['tag_ID'] ) ) {
+				foreach ( $this->get_attributes( $model ) as $attribute_name => $attribute ) {
```

Comment:
This loop doesn't seem to be needed, if you did this for the initial Add Category form then it will never have category-specific attributes at this time because the term is not created yet so it's impossible that there will be attributes assigned to it. Please use the same approach that we used for filling the Deafault Sorting drop-down in settings (in the add_settings callback here).

Also, please use the same naming, as you can see the option name changed on merge (hp_default_order).

Reply:
If I correctly understand from your previous comments the `isset( $_REQUEST['tag_ID']` is needed to show drop down with attributes when create a new category. But we need drop down with attributes only on the edit category page. Please check comments again and make changes